nix/static-binary.sh

```diff
@@ -43,12 +43,25 @@ while [ $# -ne 0 ]; do
       USE_HEAD="on"
       ;;
     --with-plugin=*)
-      plugins+=("${optarg}")
+      plugins+=("${optarg%/}")
       ;;
   esac
   shift
 done
 
+plugin_version() {
+  local plugin="$1"
+  local name="${plugin##*/}"
+  local key="VAST_PLUGIN_${name^^}_REVISION"
+  local value="$(git -C "${plugin}" rev-list --abbrev-commit --abbrev=10 -1 HEAD -- "${plugin}")"
```

Comment:
The last argument seems unnecessary? At least in a local test, this version was not able to pick up the correct flags for the analyzer plugin:

```
benno@bourbaki:~/src/vast/reviews/nix$ ./static-binary.sh --with-plugin=../examples/plugins/analyzer
nix (Nix) 2.3.10
nix-prefetch-github, version 4.0.3
rev is e5767db6626854b9d77e1869ab96ccafbe64b7f8
running nix-build --no-out-link -E \'with import /home/benno/src/vast/reviews/nix {};
  pkgsStatic."vast".override {
    versionOverride = "2021.07.29-rc1-4-ge5767db662-dirty";
    withPlugins = [ /home/benno/src/vast/reviews/plugins/broker /home/benno/src/vast/reviews/plugins/pcap ../examples/plugins/analyzer ];
    extraCmakeFlags = [  "-DVAST_PLUGIN_BROKER_REVISION=da276dfc53" "-DVAST_PLUGIN_PCAP_REVISION=da276dfc53" "-DVAST_PLUGIN_ANALYZER_REVISION=" ];
  }\'
[...]
```

Reply:
The last argument is necessary, because we want to get the hash of the most recent commit that modified any file below that path. If the path is omitted we would get the equivalent of `git rev-parse HEAD`.

Regarding the failure: The script (and also our CMake) only work with absolute paths to plugins. I'll add support for relative paths to this script. They will be interpreted relative to CWD of the caller.

Reply:
Hm, `git rev-parse HEAD` sounds like a reasonable "revision" for a plugin, I can imagine that it will depend on headers outside of the specific folder (like all of our current plugins) so just the latest change of the plugin itself may be ambiguous. But that's not an issue we need to solve right now.